bpf/bpf_sock.c

```diff
@@ -156,10 +156,12 @@ static __always_inline int sock4_update_revnat(struct bpf_sock_addr *ctx,
 }
 
 static __always_inline bool
-sock4_skip_xlate(struct lb4_service *svc, __be32 address)
+sock4_skip_xlate(struct lb4_service *svc, __be32 address __maybe_unused)
 {
 	if (lb4_to_lb6_service(svc))
 		return true;
+
+#ifndef DISABLE_EXTERNAL_IP_MITIGATION
 	if (lb4_svc_is_external_ip(svc) ||
 	    (lb4_svc_is_hostport(svc) && !is_v4_loopback(address))) {
```

Comment:
We only want to target the `lb4_svc_is_external_ip(svc)` check though, right? The `lb4_svc_is_hostport(svc)` check should not be in scope?

Also - does the IPv6 path need the same change?

Reply:
Telling the the truth, I was not aware of ipv6, cause we're ipv4 only currently. But I will check this and will get back.

Reply:
@julianwiedmann , you're right for both notes. We should only target for `lb4_svc_is_external_ip(svc)`. And we should apply the same for IPv6 aswell. Fixed.